flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java

```diff
@@ -46,25 +59,10 @@ public static Country getTerminalCountry(Terminal terminal) {
         return optionalCountry.get();
     }
 
-    static String getLoopFlowIdFromCountry(Network network, String identifiableId) {
-        Identifiable<?> identifiable = network.getIdentifiable(identifiableId);
-        if (identifiable instanceof Injection) {
-            return getLoopFlowIdFromCountry(getInjectionCountry((Injection<?>) identifiable));
-        }
-        throw new PowsyblException(String.format("Identifiable %s must be an Injection", identifiableId));
-    }
-
-    static Map<String, Integer> getIndex(List<String> idList) {
+    public static Map<String, Integer> getIndex(List<String> idList) {
         return IntStream.range(0, idList.size())
             .boxed()
-            .collect(Collectors.toMap(
-                idList::get,
-                Function.identity()
-            ));
-    }
-
-    public static Country getInjectionCountry(Injection<?> injection) {
-        return getTerminalCountry(injection.getTerminal());
+            .collect(Collectors.toMap(idList::get, Function.identity()));
     }
 
     public static List<Branch> getAllValidBranches(Network network) {
```

Comment:
Is the TODO from `.filter(NetworkUtil::isInMainSynchronousComponent)` still valid? It is from 2022

Reply:
Yes, I added some tests to highlight this. I will also write documentation